tests/python/contrib/test_hexagon/test_meta_schedule.py

```diff
@@ -62,7 +64,7 @@ def main(  # type: ignore  # pylint: disable=no-self-argument
 
 @tvm.testing.requires_hexagon
 def test_builder_runner(hexagon_launcher):
-    if hexagon_launcher._serial_number == "simulator":
+    if android_serial_number()[0] == "simulator":
```

Comment:
To a casual reader, there's nothing that connects the `android_serial_number` function to either the `hexagon_launcher` or the `hexagon_session`.  The current implementation also has incorrect behavior when running tests on multiple devices/simulators.  If `ANDROID_SERIAL_NUMBER=abcd1234,simulator`, the test would be erroneously run on the simulator.  If `ANDROID_SERIAL_NUMBER=simulator,abcd1234`, the test would be erroneously skipped on the device.

Instead of requiring user code to call a free function, can we instead have `HexagonLauncher.is_simulator()` and/or `HexagonSession.is_simulator()` methods?  That way, it's clear to the user that this is specific to the connected device, and not something that is globally true.

Reply:
Makes sense. The only issue here is that hexagon.session doesn't have access to serial number. We could pass serial number as input to the function and add is_simulator() function call

Reply:
I added is_simulator to both Session and Launcher

Reply:
That makes sense, and thank you!  That looks a lot nicer in the usage.